spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinRestTemplateCustomizer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.zipkin;
+
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Customizes {@link RestTemplate} used to send Spans to Zipkin
+ *
+ * @author Marcin Grzejszczak
+ *
+ * @since 1.1.0
+ */
+public interface ZipkinRestTemplateCustomizer {
```

Comment:
why we are making an interface vs a qualifier on `ClientHttpRequestInterceptor` should be documented, if retained, but really it shouldn't exist.

As mentioned before, I don't like adding new interfaces to expose activity that's already abstracted.

This is a wider interface than necessary, makes re-using `ClientHttpRequestInterceptor` implementations more difficult, and makes searching for help harder. For example, if I google `ZipkinRestTemplateCustomizer`, I'll find nothing. If I google `ClientHttpRequestInterceptor` I'll find immediately how to do anything from authenticate to log outbound requests.